webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ComplianceApi.scala

```diff
@@ -469,6 +519,34 @@ class ComplianceAPIService(
     }
   }.toBox
 
+  def getDirectiveCompliance(directiveId: DirectiveId, level: Option[Int]): Box[ByDirectiveRuleCompliance] = {
+    for {
+      rules    <- rulesRepo.getAll()
+      allGroups <- nodeGroupRepo.getAllNodeIds()
+      allNodeInfos <- nodeInfoService.getAll()
+      directive <- directiveRepo.getDirective(directiveId.uid)
+      releventRules = rules.filter(_.directiveIds.contains(directiveId))
+      releventNodes = rules.map(rule => (rule, RoNodeGroupRepository.getNodeIds(allGroups, rule.targets, allNodeInfos)))
+
+      byRules <- getByRulesCompliance(releventRules, level)
+//      byNodes <- getByNodesCompliance(None)
+
+//      reportsNodes = byNodes.filter(n => releventNodes.flatMap(_._2).contains(n.id))
+
+      compliance   <- getGlobalComplianceMode().toIO
+
+    } yield {
+      ByDirectiveRuleCompliance(
+        directiveId,
+        directive.map(_.id.uid.value).getOrElse("Unknown"),
+        ComplianceLevel.sum(byRules.map(_.compliance)),
+        compliance.mode,
+        byRules,
```

Comment:
Here we expect `Seq[ByDirectiveByRuleComponentCompliance]` but `getByRulesCompliance` return `Seq[ByRuleRuleCompliance]`